packages/@lwc/ssr-compiler/src/__tests__/transmogrify.spec.ts

```diff
@@ -0,0 +1,130 @@
+import { parseModule } from 'meriyah';
+import { generate } from 'astring';
+import { beforeAll } from 'vitest';
+import { transmogrify } from '../transmogrify';
+import type { Program as EsProgram } from 'estree';
+
+const COMPILED_CMP = `
```

Comment:
This feels like it's gonna get outdated real fast. 🤷

Reply:
Agreed, could we just have placeholder/stubs here instead? Literally just an `async function* tmpl() {}` should be enough to test the basics of transmogrification. Alternatively, call the compiler itself to generate this string.

Reply:
Although now that I look at it... this is pretty hello-world-ish. So maybe it's fine.

Reply:
Yeah, I took the most basic fixture we had, modified the customer code to add things like customer-written async generator functions, compiled, and stripped out the parts of the compiled output that I didn't really care about. What is left is not actually runnable code, but retains the pieces that I wanted to make assertions about.

Reply:
It might be worth adding some of these customer-written async generator things (or whatever) into a fixture somewhere and actually consuming that stuff programmatically in a way that would effect rendering. Those might be useful additional guardrails that would be a bit more "evergreen" compared to this snippet of code.